src/analyzer/analyzerbeats.cpp

```diff
@@ -15,7 +15,7 @@
 #include "track/track.h"
 
 // static
-QList<mixxx::AnalyzerPluginInfo> AnalyzerBeats::availablePlugins() {
+const QList<mixxx::AnalyzerPluginInfo> AnalyzerBeats::availablePlugins() {
```

Comment:
const does not really make sense when returning by value. 
What does clang complain here.

Reply:
The calling code uses it in a range-loop, which causes a [`-Wclazy-range-loop` warning](https://github.com/KDE/clazy/blob/d6b07989e7a9211b21ecb9b9f8bd4d237bcc453e/docs/checks/README-range-loop.md). The alternative would be to always save the value to a local const variable first, then loop over that. Should I change it?

Reply:
Ah, I see there is a different issue. We have wrong calls to "availablePlugins().size()", because availablePlugins() isn't a reference. The later call is performed on a new QList which can have a different size in theory.  
If we solve this, I think the loop warning will disappear anyway.

Reply:
Yes, in this instance it's probably better to save it to a const variable first, then check if it's empty and loop over that if not. good catch.